src/components/CompareResults/CompareResultsViewInit.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+
+import { useParams } from 'react-router-dom';
+
+import { repoMap } from '../../common/constants';
+import { useAppDispatch, useAppSelector } from '../../hooks/app';
+import useFetchCompareResults from '../../hooks/useFetchCompareResults';
+import { setSelectedRevisions } from '../../reducers/SelectedRevisions';
+import { fetchCompareResults } from '../../thunks/compareResultsThunk';
+import { fetchSelectedRevisions } from '../../thunks/selectedRevisionsThunk';
+
+// component to fetch set selectedRevs revisions when Compare View is loaded
+function CompareResultsViewInit() {
+  const dispatch = useAppDispatch();
+  const selectedRevisions = useAppSelector(
+    (state) => state.selectedRevisions.revisions,
+  );
+  const selectedRevs = selectedRevisions.map((item) => item.revision);
+
+  const { repos, revs } = useParams();
+  let paramRepos: string[];
+  let paramRevs: string[];
+  if (repos && revs) {
+    paramRepos = repos.split(',');
+    paramRevs = revs.split(',');
+  }
+
+  const { dispatchFetchCompareResults } = useFetchCompareResults();
+
+  const [newSelected, setNewSelected] = useState([]);
+
+  useEffect(() => {
+    let isSubscribed = true;
+
+    const fetchData = async (repo: string, rev: string) => {
+      const data = await dispatch(fetchSelectedRevisions({ repo, rev }));
+
+      if (isSubscribed) {
+        setNewSelected((prev) => [...prev, data.payload]);
```

Comment:
if the problem is the order, why don't you use a Map to store the results? Then the order would be in `selectedRevs`?